x-pack/plugins/triggers_actions_ui/public/types.ts

```diff
@@ -802,14 +803,14 @@ export interface ConnectorServices {
 }
 
 export interface RulesListFilters {
```

Comment:
do we really want to put everything as conditional attributes? It feels weird!

Reply:
Its actually odd that `RulesListFilters` wasn't all conditional since the `loadRulesWithKueryFilter` has all of its params marked as optional. 

If I don't use optional here then in https://github.com/elastic/kibana/pull/176456/files#diff-8cd0ce22105ab63d0b2a2d5c45c4afe34eb2c95fbda327092dfabb049d23cff0R78, we need to have bunch of empty arrays for no reason.